draft-irtf-cfrg-frost.md

```diff
@@ -857,46 +892,10 @@ parameters, to check that the signature share is valid using the following proce
     return l == r
 ~~~
 
-If any signature share fails to verify, i.e., if verify_signature_share returns False for
-any participant share, the Coordinator MUST abort the protocol for correctness reasons
-(this is true regardless of the size or makeup of the signing set selected by
-the Coordinator).
-Excluding one participant means that their nonce will not be included in the joint response `z`
-and consequently the output signature will not verify. This is because the
-group commitment will be with respect to a different signing set than the
-aggregated response.
-
-Otherwise, if all shares from participants that participated in Rounds 1 and 2 are valid, the Coordinator
-performs the `aggregate` operation and publishes the resulting signature.
-
-~~~
-  Inputs:
-  - group_commitment, the group commitment returned by compute_group_commitment,
-    an Element in G.
-  - sig_shares, a set of signature shares z_i, Scalar values, for each participant,
-    of length NUM_PARTICIPANTS, where MIN_PARTICIPANTS <= NUM_PARTICIPANTS <= MAX_PARTICIPANTS.
-
-  Outputs: (R, z), a Schnorr signature consisting of an Element R and Scalar z.
-
-  def aggregate(group_commitment, sig_shares):
-    z = 0
-    for z_i in sig_shares:
-      z = z + z_i
-    return (group_commitment, z)
-~~~
-
-The output signature (R, z) from the aggregation step MUST be encoded as follows
-(using notation from {{Section 3 of TLS}}):
-
-~~~
-  struct {
-    opaque R_encoded[Ne];
-    opaque z_encoded[Ns];
-  } Signature;
-~~~
-
-Where Signature.R_encoded is `G.SerializeElement(R)` and Signature.z_encoded is
-`G.SerializeScalar(z)`.
+The Coordinator can verify each signature share before first aggregating and verifying the
+signature under the group public key. However, since the aggregate signature is valid if
+and only if all signature shares are valid, this order of operations is more expensive
```

Comment:
the aggregate signature is valid if and only if all signature shares are valid (and subsets of signature shares will themselves not define a valid signature)